src/Runtime/Runtime/System.Windows/FrameworkElement_HandlingSizeAndAlignment.cs

```diff
@@ -77,7 +77,7 @@ internal static void INTERNAL_InitializeOuterDomElementWidthAndHeight(FrameworkE
                 // Width:
                 if (!double.IsNaN(element.Width))
                     style.width = element.Width.ToInvariantString() + "px";
-                else if (element.HorizontalAlignment == HorizontalAlignment.Stretch && !(element.INTERNAL_VisualParent is Canvas) && !(element is CheckBox))
+                else if (element.HorizontalAlignment == HorizontalAlignment.Stretch && !(element is CheckBox))
```

Comment:
Removing the check for a parent Canvas will stretch elements inside a canvas, so it is probably safer to explicitly check for a nested canvas. This can be applied for the Height as well I think.
```suggestion
                else if (element.HorizontalAlignment == HorizontalAlignment.Stretch && !(element is CheckBox) &&
                (!(element.INTERNAL_VisualParent is Canvas) || element is Canvas))
```